zeppelin-web/src/app/notebook/paragraph/result/result.controller.js

```diff
@@ -488,22 +489,7 @@ function ResultCtrl($scope, $rootScope, $route, $window, $routeParams, $location
   };
 
   const checkAndReplaceCarriageReturn = function(str) {
-    if (/\r/.test(str)) {
-      let newGenerated = '';
-      let strArr = str.split('\n');
-      for (let str of strArr) {
-        if (/\r/.test(str)) {
-          let splitCR = str.split('\r');
-          newGenerated += splitCR[splitCR.length - 1] + '\n';
-        } else {
-          newGenerated += str + '\n';
-        }
-      }
-      // remove last "\n" character
-      return newGenerated.slice(0, -1);
-    } else {
-      return str;
-    }
```

Comment:
Reason implementation of `checkAndReplaceCarriageReturn` moved to new class is for testability.
Controller method is hard to test. Unless method is referenced by `$scope`